processor/internal/transformer/destination_transformer/embedded/warehouse/uploader.go

```diff
@@ -0,0 +1,148 @@
+package warehouse
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"path"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/filemanager"
+
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stringify"
+
+	wtypes "github.com/rudderlabs/rudder-server/processor/internal/transformer/destination_transformer/embedded/warehouse/internal/types"
+	"github.com/rudderlabs/rudder-server/processor/types"
+)
+
+func (t *Transformer) CompareResponsesAndUpload(ctx context.Context, events []types.TransformerEvent, legacyResponse types.Response) {
+	if t.loggedSamplesUploader == nil {
```

Comment:
In which case are we not setting `loggedSamplesUploader`?

Reply:
In case of error.
```
        var err error
	t.loggedSamplesUploader, err = getSamplingUploader(t.conf, t.logger)
	if err != nil {
		t.logger.Errorn("Failed to create wt sampling file manager", obskit.Error(err))
	}
```